packages/hash/frontend/src/blocks/page/usePortals.tsx

```diff
@@ -50,11 +53,26 @@ export const usePortals = () => {
     setPortals(blankPortals);
   }, []);
 
-  const renderedPortals = Array.from(portals.entries()).map(
-    ([target, { key, reactNode }]) => (
-      <Fragment key={key}>{createPortal(reactNode, target)}</Fragment>
-    ),
+  const groupedPortals = Array.from(portals.entries()).reduce(
```

Comment:
Have you considered segregating the state by id, instead of calculating it here?

Reply:
I tried grouping the portals in the state by id, but it made this solution more complex because we would have to pass the `draftId` to `renderPortal` invocations so we can find the map in the state object and delete it, whereas before we were just passing `null` as the `node` value and we could find it and delete it because it was a flat map.